pkg/agent/openflow/pipeline.go

```diff
@@ -47,21 +47,15 @@ const (
 	dnatTable                   binding.TableIDType = 40
 	serviceLBTable              binding.TableIDType = 41
 	endpointDNATTable           binding.TableIDType = 42
-	EmergencyEgressRuleTable    binding.TableIDType = 45
-	SecurityOpsEgressRuleTable  binding.TableIDType = 46
-	NetworkOpsEgressRuleTable   binding.TableIDType = 47
-	PlatformEgressRuleTable     binding.TableIDType = 48
-	ApplicationEgressRuleTable  binding.TableIDType = 49
+	MultiTierEgressRuleTable    binding.TableIDType = 45
```

Comment:
do we need to reserve 46-48 tables for overflow.. or at the moment we only use a single table for all tiers (except default)

Reply:
We use a single table for all tiers except default, which should be sufficient.

Reply:
What's the consideration of having a separate table for default? I'm asking because:
1. If we assume single tier table and multiple tier table could be full, we need to have a way to distribute the rules to multiple tables anyway. If we don't assume they could be full, a single table for all tiers is simpler.
2. Two rounds of conjunction calculation is less efficient, do you have performance data for single table and two tables like before? A recent performance bug https://github.com/vmware-tanzu/antrea/issues/1219 reminds me that we should simplify the pipeline as much as possible to get best performance.

I don't mean to block this feature if it targets for 0.10, but wants to see the performance impact and if it's possible to use one table. It could be improved later if single table makes sense.

Reply:
The reason why we decide to go with two tables is that we expect a skew in number of rules created in the default (application) tier and in the other tiers. In most use cases, CNP/ANP rules will be put in the application tier, as it is the default tier for policy specs that do not specify a tier. So we thought that there should be a comparable amount of rules with all other tiers combined and the default tier. 
We could definitely use a single table if the scale requirement is loose. Remember since CNP/ANP rules need to be created on different priorities, there will be some disjoint match rules. If we created all these rules in a single table, we probably would hit max flow per table limit before the priorities are all occupied.

Reply:
Could you clarify max flow per table before the priorities are all occupied? I remember there is no limit of number of flow in a table.

Reply:
It seems that you are correct... I was told there is a limit on the number of flow in a table, but upon research I found there's no hard limit other than the memory constraint. I do found http://www.openvswitch.org/support/dist-docs/ovs-vswitchd.conf.db.5.pdf states that
```
other_config : flow-limit: optional string, containing an integer, at least 0
The maximum number of flows allowed in the datapath flow table. Internally OVS will choose a
flow limit which will likely be lower than this number, based on real time network conditions.
Tweaking this value is discouraged unless you know exactly what you’re doing.
The default is 200000.
```
so bottom line it can be toggled. Let's evaluate in the next release if it makes sense to put all tiers in one table.

Reply:
That's datapath flow table, I think its limit is to restrict the memory that will be occupied by datapath flows which are essentially cache. The userspace Openflow table doesn't have limit on flows per table. If there have to be a number of flows, the memory consumption is same regardless of single table or multiple tables, but single table has better performance. It's generally preferable to keep flows of same purpose in one table.

Reply:
Understood. Two tables does give us a bit more flexibility in priority assignment and reduce the chance of having to re-hash priorities. If we want to collide into single table, some minor changes need to be made.

Reply:
sure